envbuilder.go

```diff
@@ -108,6 +159,30 @@ func Run(ctx context.Context, opts options.Options) error {
 		}
 	}() // best effort
 
+	runtimeData := runtimeDataStore{
+		ContainerEnv: make(map[string]string),
+		RemoteEnv:    make(map[string]string),
+	}
+	if fileExists(opts.Filesystem, magicDir.Image()) {
+		if err = parseMagicImageFile(opts.Filesystem, magicDir.Image(), &runtimeData); err != nil {
+			return fmt.Errorf("parse magic image file: %w", err)
+		}
+		runtimeData.Image = true
+
+		// Some options are only applicable for builds.
+		if opts.RemoteRepoBuildMode {
+			opts.Logger(log.LevelDebug, "Ignoring %s option, it is not supported when using a pre-built image.", options.WithEnvPrefix("REMOTE_REPO_BUILD_MODE"))
+			opts.RemoteRepoBuildMode = false
+		}
+		if opts.ExportEnvFile != "" {
+			// Currently we can't support this as we don't have access to the
+			// post-build computed env vars to know which ones to export.
```

Comment:
Review: I couldn't think of a way to do this reliably. Maybe it's an easy fix but for now I'd like to leave it like this.

Reply:
I think that's fair. We can address this in a follow-up.

Reply:
https://github.com/coder/envbuilder/issues/366